### PR TITLE
Added pwsh az function container

### PR DESCRIPTION
**Pull request by brettmillerb**
Good afternoon

I noticed that there was no Azure Functions PowerShell container available so I used the C# .NET core container and added Pwsh 6.2.3 to it in the dockerfile.

I'm by no means fluent with docker but I have tested this and it builds successfully but there is an error with setting the locale. Not sure if this is an `ENV` issue that needs to be resolved.

```bash
[106207 ms] perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = "en_US.UTF-8",
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
```

The container still functions as normal and I can run and debug a local PowerShell Azure functions.

Please let me know if there are any changes required or if there's anything I have missed.

### CODE REVIEW

File: containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "name": "Azure Functions & pwsh (.NET Core 2.2)",
+    "dockerFile": "Dockerfile",
+    "appPort": 7071,
+    // Use 'settings' to set *default* container specific settings.json values on container create. 
+    // You can edit these settings after create using File > Preferences > Settings > Remote.
+    "settings": {
+        "terminal.integrated.shell.linux": "/bin/bash"
```
Comment (TylerLeonhardt) [inline containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json]: Maybe change to `pwsh`

😅

File: containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    // Uncomment the next line to use a non-root user. On Linux, this will prevent
+    // new files getting created as root, but you may need to update the USER_UID
+    // and USER_GID in .devcontainer/Dockerfile to match your user if not 1000.
+    // "runArgs": [ "-u", "vscode" ],
+    // Add the IDs of extensions you want installed when the container is created in the array below.
+    "extensions": [
+        "ms-azuretools.vscode-azurefunctions",
+        "ms-vscode.powershell"
+    ]
+}
```
Comment (TylerLeonhardt) [inline containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json]: Nit: newline at end of file

File: containers/azure-functions-pwsh-6.2.3/.devcontainer/dockerfile [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        -Command " \
+        \$ErrorActionPreference = 'Stop' ; \
+        \$ProgressPreference = 'SilentlyContinue' ; \
+        while(!(Test-Path -Path \$env:PSModuleAnalysisCachePath)) {  \
+            Write-Host "'Waiting for $env:PSModuleAnalysisCachePath'" ; \
+            Start-Sleep -Seconds 6 ; \
+        }"
+
+# Switch back to dialog for any ad-hoc use of apt-get
+ENV DEBIAN_FRONTEND=
```
Comment (TylerLeonhardt) [inline containers/azure-functions-pwsh-6.2.3/.devcontainer/dockerfile]: nit: newline at end of file.

File: containers/azure-functions-pwsh-6.2.3/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+1. If this is your first time using a development container, please follow the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started) to set up your machine.
+
+2. To use VS Code's copy of this definition:
+   1. Start VS Code and open your project folder.
+   2. Press <kbd>F1</kbd> select and **Remote-Containers: Add Development Container Configuration Files...** from the command palette.
+   3. Select the Azure Functions & pwsh (.NET Core 2.2) definition.
+
+3. To use latest-and-greatest copy of this definition from the repository:
+   1. Clone this repository.
+   2. Copy the contents of `containers/azure-functions-dotnetcore-latest/.devcontainer` to the root of your project folder.
```
Comment (TylerLeonhardt) [inline containers/azure-functions-pwsh-6.2.3/README.md]: this says dotnetcore

File: containers/azure-functions-pwsh-6.2.3/README.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+5. Finally, press <kbd>F1</kbd> and run **Remote-Containers: Reopen Folder in Container** to start using the definition.
+
+## Testing the definition
+
+This definition includes some test code that will help you verify it is working as expected on your system. Follow these steps:
+
+1. If this is your first time using a development container, please follow the [getting started steps](https://aka.ms/vscode-remote/containers/getting-started) to set up your machine.
+2. Clone this repository.
+3. Start VS Code, press <kbd>F1</kbd>, and select **Remote-Containers: Open Folder in Container...**
+4. Select the `containers/azure-functions-dotnetcore-latest` folder.
```
Comment (TylerLeonhardt) [inline containers/azure-functions-pwsh-6.2.3/README.md]: dotnetcore

**Comment by Chuxel** [comment]
@brettmillerb Thanks so much for the contribution! That's awesome. 

@TylerLeonhardt makes some great suggestions.

One other that I would add is we're typically not specifying the full version number in the dev container name unless we intend there to be more than one.  Even then, it typically is a major release or version number that breaks compatibility.  e.g. for .NET, that's 2.1.x, 2.2.x, 3.0.x, etc.  For Node.js, its 10.x.x, 12.x.x. This allows people to still get back to the README when the minor or break-fix part of the version number is increased.

Perhaps this should be `containers/azure-functions-pwsh-6` ?

File: containers/azure-functions-pwsh-6.2.3/README.md (hunk 4 above)
Comment (brettmillerb) [reply containers/azure-functions-pwsh-6.2.3/README.md]: Changed to `containers/azure-functions-pwsh-6/.devcontainer`

File: containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json (hunk 1 above)
Comment (brettmillerb) [reply containers/azure-functions-pwsh-6.2.3/.devcontainer/devcontainer.json]: Changed to `/usr/bin/pwsh`

**Comment by brettmillerb** [comment]
Made the recommendations by @TylerLeonhardt and renamed the parent folder to major version. Next version will be .NET core 3.0 and powershell 7 GA hopefully.

Let me know if there's anything else @Chuxel 🙂

**Comment by Chuxel** [comment]
Looks good - One quick question. Given .NET 2.2 is EOL Jan 1st 2020, should this use the LTS 2.1 SDK?

**Comment by TylerLeonhardt** [comment]
@Chuxel the PowerShell worker [targets netcoreapp2.2](https://github.com/Azure/azure-functions-powershell-worker/blob/dev/src/Microsoft.Azure.Functions.PowerShellWorker.csproj#L9) so I don't think we can do that.

**Comment by Chuxel** [comment]
Thanks for the clarification!  Merging.

**Comment by bergmeister** [comment]
@TylerLeonhardt @brettmillerb  Since the PowerShell workes uses .Net Core 2.1 (see [here](https://github.com/Azure/azure-functions-powershell-worker/blob/dev/src/Microsoft.Azure.Functions.PowerShellWorker.csproj#L9)), 2.2 does not make any sense and has been deprecated in December. The template should've really targeted 2.1

**Pull request merged**